src/pages/home/report/ReportActionsList.js

```diff
@@ -189,12 +190,10 @@ function ReportActionsList({
             return;
         }
 
-        if (ReportUtils.isUnread(report)) {
-            if (scrollingVerticalOffset.current < MSG_VISIBLE_THRESHOLD) {
-                Report.readNewestAction(report.reportID);
-            } else {
-                readActionSkipped.current = true;
-            }
+        if (Visibility.isVisible() && ReportUtils.isUnread(report) && scrollingVerticalOffset.current < MSG_VISIBLE_THRESHOLD) {
+            Report.readNewestAction(report.reportID);
+        } else {
+            readActionSkipped.current = true;
```

Comment:
Can you please re-add the inner condition. `readActionSkipped` should be set to true only if the report is unread.